### PR TITLE
Fix async error in class constructors.  Update docstrings

### PR DESCRIPTION
**Pull request by Jkovarik**
The constructors were sync calling an async update method.   Move call
out of constructor and into cache method, add new methods to DRY that
up and add related tests.

### CODE REVIEW

File: packages/cumulus-api-client/CumulusApiClient.js [hunk 1]
```diff
@@ -131,30 +137,65 @@ class CumulusApiClient {
...
    *     and store it inthe cache
+   * @memberof CumulusApiClient
    *
    * @returns {string} - returns active valid bearer token
    */
   async getCacheAuthToken() {
-    let token;
     try {
-      token = await this._getAuthTokenRecord();
+      let token;
```
Comment (Jkovarik) [inline packages/cumulus-api-client/CumulusApiClient.js]: This is the actual logic change, the new methods above are to DRY up/ensure updates on create calls.

File: packages/cumulus-api-client/tests/test-CumulusApiClient.js [hunk 2]
```diff
@@ -12,7 +14,7 @@ const CumulusAuthTokenError = require('../CumulusAuthTokenError');
 
 test.before(async (t) => {
   process.env.tableName = randomId('table');
-  dynamodb().createTable({
+  await dynamodb().createTable({
```
Comment (markdboyd) [inline packages/cumulus-api-client/tests/test-CumulusApiClient.js]: You can use this here instead:

https://github.com/nasa/cumulus/blob/master/packages/aws-client/DynamoDb.js#L139

File: packages/cumulus-api-client/tests/test-CumulusApiClient.js [hunk 3]
```diff
@@ -38,18 +48,30 @@ test.before(async (t) => {
...
 );
 
+test.serial('getCacheAuthToken initializes cache and sets token if cacheInitialized is false', async (t) => {
+  const token = randomId();
+  const testApiClient = new CumulusApiClientRewire(t.context.config);
+  testApiClient.createNewAuthToken = async () => token;
+  testApiClient._validateTokenExpiry = async () => true;
+  const actual = await testApiClient.getCacheAuthToken();
+  t.is(token, actual);
+  t.is(true, testApiClient.cacheInitialized);
```
Comment (markdboyd) [inline packages/cumulus-api-client/tests/test-CumulusApiClient.js]: Nit: you can do `t.true(testApiClient.cacheInitialized)`

File: packages/cumulus-api-client/tests/test-CumulusApiClient.js [hunk 4]
```diff
@@ -232,3 +262,22 @@ test.serial('get retries/creates a token the expected number of times, then thro
...
 });
+
+test.serial('_createAndUpdateNewAuthToken creates a new token and updates the cache', async (t) => {
+  const token = randomId();
+  const testApiClient = new CumulusApiClientRewire(t.context.config);
+  testApiClient._validateTokenExpiry = async () => true;
+  testApiClient.createNewAuthToken = async () => token;
+  await testApiClient._createAndUpdateNewAuthToken();
+  const actual = await testApiClient.getCacheAuthToken();
+  t.is(token, actual);
```
Comment (markdboyd) [inline packages/cumulus-api-client/tests/test-CumulusApiClient.js]: I'm probably missing something - but how does this test verify that the cache itself is updated?

File: packages/cumulus-api-client/tests/test-CumulusApiClient.js (hunk 2 above)
Comment (Jkovarik) [reply packages/cumulus-api-client/tests/test-CumulusApiClient.js]: 💯  nice

File: packages/cumulus-api-client/tests/test-CumulusApiClient.js (hunk 4 above)
Comment (Jkovarik) [reply packages/cumulus-api-client/tests/test-CumulusApiClient.js]: You're not.   Two issues: it needs to be set to initialized on setup, and the test needs to have a new mock set for 'createNewAuthToken' to prove it's not generating the same thing.     

Nice catch, I'm glad you were taking a look at this. 

5e9aaa289

Comment (Jkovarik) [reply packages/cumulus-api-client/tests/test-CumulusApiClient.js]: Better refactored: d7b38ac51

File: packages/cumulus-api-client/tests/test-CumulusApiClient.js (hunk 2 above)
Comment (Jkovarik) [reply packages/cumulus-api-client/tests/test-CumulusApiClient.js]: df015f064

File: packages/cumulus-api-client/tests/test-CumulusApiClient.js (hunk 3 above)
Comment (Jkovarik) [reply packages/cumulus-api-client/tests/test-CumulusApiClient.js]: df015f064

**Pull request merged**